🔀 :: (#36) - publish ExerciseScreen

robo710 opened the pull request:
## 💡 개요
- 운동 화면을 퍼블리싱 했습니다.
## 📃 작업내용
- 운동 화면을 퍼블리싱 했습니다.
## 🔀 변경사항
- add ExerciseScreen
- add AddExerciseScreen
## 🙋‍♂️ 질문사항
- 코드에 이상한 부분이 있다면 Comment 달아주세요.
## 🍴 사용방법

## 🎸 기타

robo710 committed:
Theme Setting

robo710 committed:
Theme Setting

robo710 committed:
Exercise Module 생성

robo710 committed:
Exercise Module 생성

robo710 committed:
필요 Icon 추가

robo710 committed:
필요 Icon 추가

robo710 committed:
Add tint in HeartIcon

robo710 committed:
ExerciseButton 생성

robo710 committed:
ExerciseButton 생성

robo710 committed:
ExerciseItem 생성

robo710 committed:
ExerciseItem 생성

robo710 committed:
ExerciseScreen 생성

robo710 committed:
ExerciseScreen 생성

robo710 committed:
AddExerciseScreen 생성

robo710 committed:
AddExerciseScreen 생성

robo710 committed:
ExerciseNavigation 생성

robo710 committed:
ExerciseNavigation 생성

robo710 committed:
TextField 수정

robo710 committed:
TextField 수정

robo710 committed:
create ExerciseNavigation

robo710 committed:
create ExerciseNavigation

robo710 committed:
pull develop

robo710 committed:
apply SSBAndroidTheme

robo710 committed:
apply SSBAndroidTheme

robo710 committed:
apply SSBAndroidTheme

robo710 committed:
apply SSBAndroidTheme

robo710 committed:
Merge branch 'develop' of https://github.com/Sweat-Soaked-Body/SSB-Android into feature/36-exercise-screen

robo710 committed:
ColorTheme 수정

robo710 committed:
ColorTheme 수정

audgns10 commented on core/design-system/src/main/java/com/sweat/design_system/icon/SSBIcon.kt:
```diff
@@ -32,6 +32,16 @@ fun CheckIcon(modifier: Modifier = Modifier) {
+        modifier = modifier
+    )
+}
+
```
함수명은 Icon으로 되어있지만 아래 컴포저블은 Icon이 아닌 Image로 사용하고 있어요.

audgns10 replied:
수정부탁드립니다.

robo710 replied:
수정했습니다

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+    val (exerciseTextState, onExerciseTextChange) = remember { mutableStateOf("") }
+    val exerciseStyleExpanded = remember { mutableStateOf(false) }
+    val exerciseStyle = listOf("시간", "세트")
+    val selectedStyle = remember { mutableStateOf("") }
```
state만들어서 처리해주세요

audgns10 commented on core/design-system/src/main/java/com/sweat/design_system/icon/SSBIcon.kt:
```diff
@@ -72,6 +82,26 @@ fun ChevronRightIcon(modifier: Modifier = Modifier) {
+        modifier = modifier
+    )
+}
+
```
해당 부분도 위 코맨트와 같습니다. 수정부탁드립니다.

robo710 replied:
수정했습니다

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+                        .padding(1.dp)
+                        .width(24.dp)
+                        .height(24.dp)
+                        .clickable(onClick = { /*TODO*/ })
```
싱글클릭커블 함수 써주세요

robo710 replied:
수정했습니다

answad commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+            )
+        }
+    }
+}
```
이쪽 컴포넌트로 분리 해주세요

robo710 replied:
수정했습니다

audgns10 commented on feature/exercise/src/main/java/com/sweat/exercise/ExerciseNavigation.kt:
```diff
@@ -0,0 +1,23 @@
+            navigateToAddExerciseScreen = navigateToAddExerciseScreen
+        )
+    }
+}
```
해당 디자인을 보니 popUpBackStack() 처리를 해주어야할 것 같아여! 추가해주세요!

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+            Row(
+                modifier = modifier
+                    .fillMaxWidth()
+                    .height(57.dp)
```
반응형으로 짜주세요

answad commented on feature/exercise/src/main/java/com/sweat/exercise/view/ExerciseScreen.kt:
```diff
@@ -0,0 +1,188 @@
+@Composable
+fun ExerciseScreen(
+    modifier: Modifier = Modifier,
+    exerciseItems: List<Pair<String, Boolean>>
```
list 는 compose 컴파일러가 불안정하다고 여기기때문에 ImmutableList 사용해주세요

robo710 replied:
수정했습니다

audgns10 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+                    modifier = modifier
+                        .padding(1.dp)
+                        .width(24.dp)
+                        .height(24.dp)
```
size 사용해도 될 것 같아여!
padding(1.dp)는 있는 이유가 있나요?

robo710 replied:
디자인에 padding(1.dp)가 포함되어 있어서 넣었습니다
size로 수정하였습니다

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+                    .fillMaxWidth()
+                    .padding(horizontal = 24.dp)
+            ) {
+                Text(
```
컴포넌트로 만들어주세요

audgns10 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+                        .padding(1.dp)
+                        .width(24.dp)
+                        .height(24.dp)
+                        .clickable(onClick = { /*TODO*/ })
```
인자에 BackStack 처리를 할 수 있게 추가해서 클릭시 이벤트 처리를 해주세요!

audgns10 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+                )
+                Text(
+                    text = "운동 추가",
+                    style = typography.subTitle
```
텍스트 컬러 추가 부탁드립니다!

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+                            modifier = Modifier
+                                .fillMaxWidth()
+                                .border(1.dp, Color.LightGray, RoundedCornerShape(8.dp))
+                                .padding(vertical = 5.dp)
```
패딩이 잘못 들어가 있습니다

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+                                Row(
+                                    modifier = Modifier
+                                        .fillMaxWidth()
+                                        .clickable {
```
싱글 클릭커블 사용해주세요

robo710 replied:
수정했습니다

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+                                            selectedCategory.value = category
+                                            exerciseTypeExpanded.value = false
+                                        }
+                                        .padding(horizontal = 16.dp, vertical = 9.dp),
```
버티컬 패딩이 잘못 들어가 있습니다

robo710 replied:
수정했습니다

audgns10 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+                                        ChevronSmallUpLightIcon(
+                                            modifier = Modifier
+                                                .width(24.dp)
+                                                .height(24.dp)
```
size를 사용해주어도 될 것 같아여!

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+                                .height(54.dp)
+                                .border(
+                                    width = 1.dp,
+                                    color = Color(0xFFEFF0F2),
```
16진수 쓰지 말아주세요

robo710 replied:
수정했습니다

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+                                    color = Color(0xFFEFF0F2),
+                                    shape = RoundedCornerShape(size = 8.dp)
+                                )
+                                .padding(start = 16.dp, top = 14.dp, end = 16.dp, bottom = 14.dp),
```
vertical, horizontal 사용해주세요

robo710 replied:
수정했습니다

audgns10 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+                                    color = Color(0xFFEFF0F2),
+                                    shape = RoundedCornerShape(size = 8.dp)
+                                )
+                                .padding(start = 16.dp, top = 14.dp, end = 16.dp, bottom = 14.dp),
```
줄바꿈 해주시고, 각각 horizontal / vertical로 처리할 수 있을 것 같아요!

audgns10 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+                        ) {
+                            Text(
+                                text = selectedCategory.value,
+                                style = typography.bodySmall
```
색상 추가해주세요!

robo710 replied:
수정했습니다!

audgns10 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+                            ChevronSmallDownLightIcon(
+                                modifier = modifier
+                                    .width(24.dp)
+                                    .height(24.dp)
```
size로 해주셔도 될 것 같아요!

robo710 replied:
수정했습니다!

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+                        }
+                    }
+                }
+                Spacer(modifier = modifier.height(20.dp))
```
스페이서 값이 잘못들어가 있습니다

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+                    modifier = modifier
+                        .border(
+                            width = 1.dp,
+                            color = Color(0xFFEFF0F2),
```
16진수 쓰지 말아주세요

robo710 replied:
수정했습니다

audgns10 commented on feature/exercise/src/main/java/com/sweat/exercise/view/component/ExerciseItem.kt:
```diff
@@ -0,0 +1,68 @@
+                .background(Color.White)
+                .padding(
+                    vertical = 17.dp,
+                ),
```
한 줄로 수정해주세요!

robo710 replied:
수정했습니다!

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+                            shape = RoundedCornerShape(size = 8.dp)
+                        )
+                        .fillMaxWidth()
+                        .height(54.dp)
```
반응형으로 구현해주세요

audgns10 commented on feature/exercise/src/main/java/com/sweat/exercise/view/component/ExerciseItem.kt:
```diff
@@ -0,0 +1,68 @@
+                isSelected = isSelected
+            )
+        }
+        Divider(thickness = 1.dp, color = colors.gray100)
```
줄 바꿈 해주세요!

robo710 replied:
수정했습니다

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+                        modifier = modifier.fillMaxWidth(),
+                        value = exerciseTextState,
+                        onValueChange = { newText -> onExerciseTextChange(newText) },
+                        textStyle = typography.bodySmall.copy(
```
줄바꿈 하지 말아주세요

robo710 replied:
수정했습니다

audgns10 commented on feature/exercise/src/main/java/com/sweat/exercise/view/component/ExerciseItem.kt:
```diff
@@ -0,0 +1,68 @@
+            Spacer(modifier = modifier.weight(1f))
+            HeartIcon(
+                modifier = modifier
+                    .padding(1.dp)
```
패딩이 왜 필요한지 설명해주세요.

robo710 replied:
디자인에 패딩이 있는 것으로 봐서 추가했습니다

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+                        }
+                    )
+                }
+                Spacer(modifier = modifier.height(20.dp))
```
스페이서 값이 잘못들어가 있습니다

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+                                Row(
+                                    modifier = Modifier
+                                        .fillMaxWidth()
+                                        .clickable {
```
싱글클릭커블 함수 사용해주세요

robo710 replied:
수정했습니다

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+                                            selectedStyle.value = exerciseStyle
+                                            exerciseStyleExpanded.value = false
+                                        }
+                                        .padding(horizontal = 16.dp, vertical = 9.dp),
```
패딩값이 잘못 들어가 있습니다

audgns10 reviewed (commented):
중첩클릭을 막기위해 구현한 clickableSingle 함수를 clickable 대신 사용해주세요!

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+                                .height(54.dp)
+                                .border(
+                                    width = 1.dp,
+                                    color = Color(0xFFEFF0F2),
```
16진수 사용하지 말아주세요

robo710 replied:
수정했습니다

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,302 @@
+                                    color = Color(0xFFEFF0F2),
+                                    shape = RoundedCornerShape(size = 8.dp)
+                                )
+                                .padding(start = 16.dp, top = 14.dp, end = 16.dp, bottom = 14.dp),
```
vertival, horizontal 사용해주세요

robo710 replied:
수정했습니다

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/ExerciseScreen.kt:
```diff
@@ -0,0 +1,188 @@
+    val selectedButton = remember { mutableStateOf(exerciseList.first()) }
+    val exerciseStateList = remember { mutableStateListOf<Pair<String, Boolean>>().apply { addAll(exerciseItems) } }
+    val isSearching = remember { mutableStateOf(false) }
+    val (searchTextState, onSearchTextChange) = remember { mutableStateOf("") }
```
state사용해주세요

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/ExerciseScreen.kt:
```diff
@@ -0,0 +1,188 @@
+            Row(
+                modifier = modifier
+                    .fillMaxWidth()
+                    .height(57.dp)
```
반응형으로 구현해주세요

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/ExerciseScreen.kt:
```diff
@@ -0,0 +1,188 @@
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                if (isSearching.value) {
+                    BasicTextField(
```
컴포넌트로 빼주세요

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/ExerciseScreen.kt:
```diff
@@ -0,0 +1,188 @@
+                        modifier = modifier.fillMaxWidth(),
+                        value = searchTextState,
+                        onValueChange = { newText -> onSearchTextChange(newText) },
+                        textStyle = typography.bodySmall.copy(
```
줄바꿈 하지 말아주세요

robo710 replied:
수정했습니다

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/ExerciseScreen.kt:
```diff
@@ -0,0 +1,188 @@
+                    )
+                    Spacer(modifier = modifier.weight(1f))
+                    Row(
+                        modifier = modifier
```
반응형으로 만들어주세요

robo710 replied:
수정했습니다

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/ExerciseScreen.kt:
```diff
@@ -0,0 +1,188 @@
+                    ) {
+                        PlusIcon(
+                            modifier = modifier
+                                .padding(1.dp)
```
디자인에서 패딩이 들어가 있지 않습니다

robo710 replied:
수정했습니다

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/ExerciseScreen.kt:
```diff
@@ -0,0 +1,188 @@
+                        )
+                        SearchIcon(
+                            modifier = modifier
+                                .padding(1.dp)
```
디자인에서 패딩이 들어가 있지 않습니다

robo710 replied:
수정했습니다

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/ExerciseScreen.kt:
```diff
@@ -0,0 +1,188 @@
+            LazyColumn(
+                modifier = modifier
+                    .fillMaxWidth()
+                    .fillMaxHeight()
```
fillMaxSize 사용해주세요

robo710 replied:
수정했습니다

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/component/ExerciseButton.kt:
```diff
@@ -0,0 +1,79 @@
+        modifier = modifier
+            .border(
+                width = 1.dp,
+                color = if (state == ButtonState.Enabled) Color(0xFFEFF0F2) else Color.Black,
```
16진수 사용하지 말아주세요

robo710 replied:
수정했습니다

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/component/ExerciseItem.kt:
```diff
@@ -0,0 +1,68 @@
+            HeartIcon(
+                modifier = modifier
+                    .padding(1.dp)
+                    .width(24.dp)
```
size사용해주세요

robo710 replied:
수정했습니다

oheunchan07 commented on feature/exercise/src/main/java/com/sweat/exercise/view/component/ExerciseItem.kt:
```diff
@@ -0,0 +1,68 @@
+                    .padding(1.dp)
+                    .width(24.dp)
+                    .height(24.dp)
+                    .clickable { onHeartClick() },
```
싱글클릭커블 함수 사용해주세요

robo710 replied:
수정했습니다

audgns10 committed:
:sparkles: :: Create GenderEnum

audgns10 committed:
:sparkles: :: Create SignUpViewModel

audgns10 committed:
:memo: :: Add Fail State In "catch" <- LoginViewModel

audgns10 committed:
:recycle: :: Add keyboardOptions

audgns10 committed:
:recycle: :: Refactor Manage State In ViewModel / Apply KeyboardOptions

audgns10 committed:
:fire: :: Fire Unused Code In SignUpNavigation

audgns10 committed:
:memo: :: Apply Before Commit

audgns10 committed:
:fire: :: Fire Unused Files

audgns10 committed:
:truck: :: Add navigation Package And Add SignUpNavigation

audgns10 committed:
:truck: :: Renaming

audgns10 committed:
:recycle: :: Refactor Code Style

robo710 committed:
Add ExerciseSelector

robo710 committed:
Add ExerciseTextField

robo710 committed:
Change Image to Icon

robo710 committed:
apply ExerciseTextField & ExerciseSelector

robo710 committed:
change Text color and padding

robo710 committed:
change size and padding

robo710 committed:
change color

robo710 committed:
include exercise feature

robo710 committed:
add selectedItem, expanded parameter

robo710 committed:
apply selectedItem, expanded parameter

robo710 committed:
change list to ImmutableList

robo710 committed:
add ExerciseViewModel

robo710 committed:
rename ExerciseViewModel to AddExerciseViewModel

robo710 committed:
add dependencies

robo710 committed:
Adjust size and line break settings

robo710 committed:
Adjust size and Change height

robo710 committed:
Adjust size and line break settings

robo710 committed:
remove unused import

answad committed:
feat: 친구 추가 성공 카드를 위한 Composable 추가

- 로컬 GIF 파일을 사용하여 친구 추가 완료 표시
- `AddFriendSuccessCard` 컴포저블 함수 추가
- Coil을 사용하여 GIF 이미지 로딩 및 표시

answad committed:
feat: 로컬 gif 파일 추가

answad committed:
feat: 로컬 gif 파일 을 띄우기 위한 coil과 gif 확장 의존성 추가

robo710 committed:
ExerciseSelector 수정

robo710 committed:
AddExerciseViewModel 설정

robo710 committed:
rename ExerciseSelector & ExerciseTextField

robo710 committed:
Add ExerciseTextField

robo710 committed:
rename ExerciseSelector & ExerciseTextField

robo710 committed:
rename ExerciseSelector & ExerciseTextField

robo710 committed:
rename ExerciseSelector & ExerciseTextField

robo710 committed:
add ExerciseTextField

robo710 committed:
create ExerciseViewModel

robo710 committed:
Adjust line breaks

audgns10 committed:
:fire: :: Fire Unused FIles

audgns10 committed:
:memo: :: Add .gitkeep Files

audgns10 committed:
:memo: :: Add Resource Folder -> value.strings.xml

audgns10 committed:
:fire: :: Fire Unused Dependency

audgns10 committed:
:sparkles: :: Create SignUpRequest(DTO, Param)

audgns10 committed:
:sparkles: :: Create Auth Mappers(SignUp, Login)

audgns10 committed:
:sparkles: :: Create makeToast

audgns10 committed:
:recycle: :: Refactor AuthDataSource(Impl)

audgns10 committed:
:recycle: :: Refactor AuthRepository

audgns10 committed:
:sparkles: :: Create SignUpRequestUseCase

audgns10 committed:
:sparkles: :: Add Network Logic(SignUp)

audgns10 committed:
:memo: :: Add Dependency

audgns10 committed:
:recycle: :: Add Dependency -> Refactor Duplication Code

audgns10 committed:
:sparkles: :: Add SignUp API

audgns10 committed:
:memo: :: Apply RemoteDataSourceModule

audgns10 committed:
:memo: :: Add.Navigation Code

audgns10 committed:
:sparkles: :: Create onErrorToast

audgns10 committed:
:recycle: :: Apply Before Commit

robo710 committed:
Change AddExerciseIntent name

robo710 committed:
update viewmodel

robo710 committed:
preview -> DevicePreviews

robo710 committed:
change ExercisePreview

robo710 committed:
remove unused import

robo710 committed:
change route

robo710 committed:
Add PopUpBackStack

robo710 committed:
change AddExerciseScreenSideEffect

robo710 committed:
Create AddExerciseNavigation

robo710 committed:
Create AddExerciseNavigation

robo710 committed:
AddExerciseNavigation 위치 변경

robo710 committed:
Fix merge conflicts

robo710 committed:
todo -> PopUpBackStack

robo710 committed:
Merge remote-tracking branch 'origin/feature/36-exercise-screen' into feature/36-exercise-screen

# Conflicts:
#	core/design-system/src/main/java/com/sweat/design_system/icon/SSBIcon.kt
#	feature/exercise/src/main/java/com/sweat/exercise/ExerciseNavigation.kt
#	feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt
#	feature/exercise/src/main/java/com/sweat/exercise/view/ExerciseScreen.kt
#	feature/exercise/src/main/java/com/sweat/exercise/view/component/ExerciseButton.kt
#	feature/exercise/src/main/java/com/sweat/exercise/view/component/ExerciseItem.kt

robo710 committed:
pull 오류 수정

robo710 committed:
ExerciseSreen의 PopUpBackStack 제거

robo710 committed:
SSBNavHost에 exerciseRoute & addExerciseRoute 추가

robo710 committed:
navigateToExerciseScreen -> navigateToExercise

robo710 committed:
navigateToAddExerciseScreen -> navigateToAddExercise

robo710 committed:
exerciseRoute & addExerciseRoute 설정

answad commented on feature/exercise/src/main/java/com/sweat/exercise/AddExerciseNavigation.kt:
```diff
@@ -0,0 +1,25 @@
+    composable(exerciseRoute){
+        AddExerciseRoute(
+            navigateToExerciseScreen = navigateToExerciseScreen,
+            popUpBackStack = popUpBackStack,
```
[AddExerciseNavigation.kt](https://github.com/Sweat-Soaked-Body/SSB-Android/pull/68/files#diff-b2dd402707d5aa4b33803042a5731cf9b9eb2b1154fec31f0cd407e4ad469ad6)
이 파일과 

[ExerciseNavigation.kt](https://github.com/Sweat-Soaked-Body/SSB-Android/pull/68/files#diff-ce8ccbb867f62c5beb9cd0bac462f66d4a4ba3e5fb094fd5aad1864a2fef05f1)
이파일을 합쳐주세요

answad commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,200 @@
+    val lifecycle = LocalLifecycleOwner.current.lifecycle
+
+    LaunchedEffect(lifecycle) {
+        lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED){
```
이 코드를 어떤 이유에서 사용하신건가요

answad commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,200 @@
+                    modifier = modifier
+                        .padding(1.dp)
+                        .size(24.dp)
+                        .clickableSingle(onClick = { handleIntent(AddExerciseIntent.PopUpBackStack) })
```
.clickableSingle{ handleIntent(AddExerciseIntent.PopUpBackStack) }
                        이런식으로 쓰는게 좀더 깔끔할거같아요

robo710 committed:
ChevronLeftIcon의 clickableSingle 수정

answad commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,200 @@
+                is AddExerciseIntent.ToggleExerciseStyleDropdown -> {
+                    previewState.value = previewState.value.copy(exerciseStyleExpanded = !previewState.value.exerciseStyleExpanded)
+                }
+                else-> {}
```
handleIntent = { _ ->}
        이런식으로 사용해주세여

answad commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,200 @@
+    ) }
+
+    AddExerciseScreen(
+        state = previewState.value,
```
이부분은 getInitState으로 주입해주세요

robo710 committed:
AddExerciseScreenPreview 수정

robo710 committed:
ExerciseNavigation에 AddExerciseNavigation 합치기

robo710 committed:
ExercisePreview 수정

robo710 committed:
delete unused import

robo710 committed:
delete LaunchedEffect

robo710 committed:
change route parameter

robo710 committed:
category 선택 기능 추가

answad commented on feature/exercise/src/main/java/com/sweat/exercise/ExerciseNavigation.kt:
```diff
@@ -0,0 +1,40 @@
+){
+    composable(exerciseRoute){
+        ExerciseRoute(
+        )
```
여기에 네비게이션 함수 연결해주세요

robo710 replied:
수정했습니다

answad commented on feature/exercise/src/main/java/com/sweat/exercise/viewModel/AddExerciseViewModel.kt:
```diff
@@ -0,0 +1,55 @@
+            is AddExerciseIntent.ExerciseStyle -> setState { copy(selectedStyle = intent.style) }
+            is AddExerciseIntent.ToggleExerciseTypeDropdown -> setState { copy(exerciseTypeExpanded = !exerciseTypeExpanded) }
+            is AddExerciseIntent.ToggleExerciseStyleDropdown -> setState { copy(exerciseStyleExpanded = !exerciseStyleExpanded) }
+            AddExerciseIntent.Exercise -> postSideEffect(AddExerciseScreenSideEffect.NavigateToExercise)
```
intent 이름을 navigateToExercise 로 바꿔주세요

robo710 replied:
수정했습니다

robo710 committed:
addExerciseRoute 수정

audgns10 commented on feature/exercise/src/main/java/com/sweat/exercise/ExerciseNavigation.kt:
```diff
@@ -0,0 +1,39 @@
+
+fun NavController.navigateToAddExerciseRoute(navOptions: NavOptions? = null){
+    this.navigate(addExerciseRoute, navOptions)
+}
```
NavController 네이밍을 navigateToAaExercise 이렇게 변경해주시는게 조금 더 좋아보여요.

robo710 replied:
수정했습니다

audgns10 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,157 @@
+
+@Composable
+fun AddExerciseRoute(
+    modifier: Modifier = Modifier,
```
해당 라우트 부붕에서는 Modifier 인자가 굳이 필요없을 것 같아요.

audgns10 replied:
해당 라우트 부붕에서는 Modifier 인자가 굳이 필요없을 것 같아요.

audgns10 replied:
해당 라우트 부붕에서는 Modifier 인자가 굳이 필요없을 것 같아요.

audgns10 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,157 @@
+
+@Composable
+fun AddExerciseRoute(
+    modifier: Modifier = Modifier,
```
해당 라우트 부붕에서는 Modifier 인자가 굳이 필요없을 것 같아요.

robo710 replied:
수정했습니다

audgns10 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,157 @@
+}
+
+@Composable
+fun AddExerciseScreen(
```
private fun으로 접근 제한자를 수정해주시면 더 좋을 것 같아요.

robo710 replied:
수정했습니다

audgns10 commented on feature/exercise/src/main/java/com/sweat/exercise/view/AddExerciseScreen.kt:
```diff
@@ -0,0 +1,157 @@
+import kotlinx.collections.immutable.persistentListOf
+
+@Composable
+fun AddExerciseRoute(
```
internal fun으로 접근 제한자를 수정해주시면 더 좋을 것 같아요.

robo710 replied:
수정했습니다

robo710 committed:
색상추가

robo710 committed:
접근 제어자 수정

robo710 committed:
changes

robo710 committed:
change navigateToAddExerciseRoute -> navigateToAddExercise

robo710 committed:
AddExerciseIntent.Exercise -> AddExerciseIntent.navigateToExercise

robo710 committed:
delete unused Import

answad reviewed (approved):
감사합니다
